parser.go

```diff
@@ -97,10 +97,14 @@ func nameChar(c byte) bool {
 
 // parseIdentifier parses an identifier.
 func (p *parser) parseIdentifier() (result string, err error) {
-	startingDash := false
-	if len(p.s) > p.i && p.s[p.i] == '-' {
-		startingDash = true
+	const prefix = '-'
+	var numPrefix int
+
+countPrefix:
+	if len(p.s) > p.i && p.s[p.i] == prefix {
```

Comment:
Rather than using a `goto`, I think you could just replace `if` with `for`.

Reply:
8919e381b2b9868b86bb29a833d893796a188f1f